app.js

```diff
@@ -27,8 +30,56 @@ app.post("/signup", function(req, res) {
 		email: req.body.email,
 		username: req.body.username,
 		password: req.body.password,
+	})
+	.then(function(user) {
+		req.session.userid = user.id;
+		req.redirect("/home");
+	})
+	.catch(function(err) {
+		renderTemplate(res, "Signup", "signup", {
+			error: "Invalid username or password",
+		});
 	});
 });
+
+app.get("/login", function(req, res) {
+	renderTemplate(res, "Login", "login");
+});
+
+app.post("/login", function(req, res) {
+	User.findOne({
+		where: {
+			username: req.body.username,
+		},
+	})
+	.then(function(user) {
+		if (user) {
+			user.comparePassword(req.body.password).then(function(valid) {
+				if (valid) {
+					req.session.userid = user.get("id");
+					res.redirect("/home");
```

Comment:
It looks like you don't have a `/home` route yet, so you probably don't want to redirect there.